src/cascadia/WindowsTerminal/TrayIcon.cpp

```diff
@@ -20,9 +22,41 @@ TrayIcon::TrayIcon(const HWND owningHwnd) :
 
 TrayIcon::~TrayIcon()
 {
+    DestroyWindow(_trayIconWndProc);
     RemoveIconFromTray();
 }
 
+void TrayIcon::CreateWindowProcess()
+{
+    WNDCLASSW wc{};
+    wc.hCursor = LoadCursor(nullptr, IDC_ARROW);
+    wc.hInstance = reinterpret_cast<HINSTANCE>(&__ImageBase);
+    wc.lpszClassName = L"TRAY_ICON_HOSTING_WINDOW_CLASS";
+    wc.style = CS_HREDRAW | CS_VREDRAW;
+    wc.lpfnWndProc = &TrayIcon::_WindowProc;
+    wc.hIcon = static_cast<HICON>(GetActiveAppIconHandle(true));
+    RegisterClass(&wc);
+
+    _trayIconWndProc = CreateWindowW(wc.lpszClassName,
+                                     wc.lpszClassName,
+                                     WS_DISABLED,
+                                     CW_USEDEFAULT,
+                                     CW_USEDEFAULT,
+                                     CW_USEDEFAULT,
+                                     CW_USEDEFAULT,
+                                     HWND_MESSAGE,

```

Comment:
Wasn't _actually_ sure what were the necessary options to make this window process as un-interactable as possible, but I believe `WS_DISABLED` and _maybe_ `HWND_MESSAGE` should be good?

Reply:
IIRC, `HWND_MESSAGE` is enough to make it totally uninteractable. Though, in the past we've also done:

```c++

                hwnd = CreateWindowExW(
                    0, PSEUDO_WINDOW_CLASS, nullptr, WS_OVERLAPPEDWINDOW, 0, 0, 0, 0, HWND_DESKTOP, nullptr, nullptr, nullptr);
```